ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/eea/AbstractEeaSendRawTransaction.java

```diff
@@ -108,9 +108,6 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
         final Transaction privateMarkerTransaction =
             createPrivateMarkerTransaction(privateTransaction, user);
 
-        LOG.error("CHEESE");
```

Comment:
We do have a disconnect between the privateMarkerTransactionFactory.getSender() call and the createPrivateMarkerTransaction() call. When the plugin receives the call to create the PMT it does not know what the sender was returned by the other call. 
At a minimum we need to pass in the sender to the second call.
I would prefer that the plugin does the locking if that is necessary. If you are using random keys you don't have to lock!
Besu could signal to the plugin when a PMT was un- or successfully added to the pool using a callback. Callback could be null if this is not needed.
The plugin would have to be able to get the nonce for a certain key from besu.
What do you think?

Reply:
Fair point, wired up changes to pass along the sender. 

I know we don't need locking if your using random keys, but the impact is minimal. Yes, we're locking on an address that we're not going to use again. But the lock is on a per-address basis, so shouldn't have any impact on performance.

I think it's nice to tell the plugin to go sign something, the interface and responsibilities of what I need to do as a plugin implementor are clear. I think the locking and nonce fetching should be opaque.